### PR TITLE
fix: AWS env vars to drun and vault operator

### PR DESCRIPTION
**Pull request by mojtabaimani**
fix: Adding AWS env vars to drun and vault operator

### CODE REVIEW

File: values/vault-operator/vault-operator-raw.gotmpl [hunk 1]
```diff
@@ -61,6 +61,10 @@ resources:
           value: {{ $kms.awskms.access_key }}
         - name: AWS_SECRET_ACCESS_KEY
           value: {{ $kms.awskms.secret_key }}
+        - name: AWS_REGION
+          value: {{ $kms.awskms.region }}
+        - name: AWS_DEFAULT_REGION
+          value: {{ $kms.awskms.region }}
```
Comment (Morriz) [inline values/vault-operator/vault-operator-raw.gotmpl:67]: please rewrite like this
```
{{- with $kms.awskms }}
  ...
  value: {{- .region }}
  ...
{{- end }}
```

**Review by Morriz — CHANGES_REQUESTED** [review]
I see issues this PR because @j-zimnowoda did not properly support AWS KMS with SOPS (see https://github.com/mozilla/sops#kms-aws-profiles):
- `~/.aws/credentials` is not mounted in otomi drun for SOPS
- `SOPS_KMS_ARN` needs to be injected into drun

So the context of this work item needs to be known in order to get the expected outcome, which is that SOPS as well as Vault Operator can use the 3 cloud KMSes we support: 
- AWS KMS
- GCP KMS
- Azure Vault

The other integration point is the Drone chart, which is passing the same environment variables. Please look in the `values/drone.gotmpl` file to see. I now see that he also forgot to rewrite those params there as well, as those should now be coming from the `kms:` prop.

File: values/drone/drone.gotmpl [hunk 2]
```diff
@@ -65,10 +66,34 @@ kubernetes:
     {{- tpl $addIgnorePolicyAnnotation (dict "policies" (list "psp-allowed-users" "psp-privileged" "psp-host-filesystem" "banned-image-tags")) | nindent 4 }}
   envFile: |
     {{- if $v.sops.enabled }}
-    AZURE_TENANT_ID: '{{ $v | get "azure.keyVault.tenantId" ($v | get "azure.tenantId" "") }}'
-    AZURE_CLIENT_ID: '{{ $v | get "azure.keyVault.clientId" "" }}'
-    AZURE_CLIENT_SECRET: '{{ $v | get "azure.keyVault.clientSecret" "" }}'
-    GCLOUD_SERVICE_KEY='{{ $v | get "google.kmsAccount" "" | replace "\n" "" }}'
+
+    {{- if hasKey $kms "azurekeyvault" }}
+    {{- with $kms.azurekeyvault }}
```
Comment (Morriz) [inline values/drone/drone.gotmpl:71]:
```suggestion
    {{- with ($kms | get "azurekeyvault" nil) }}
```

File: values/drone/drone.gotmpl [hunk 3]
```diff
@@ -65,10 +66,34 @@ kubernetes:
     {{- tpl $addIgnorePolicyAnnotation (dict "policies" (list "psp-allowed-users" "psp-privileged" "psp-host-filesystem" "banned-image-tags")) | nindent 4 }}
   envFile: |
     {{- if $v.sops.enabled }}
-    AZURE_TENANT_ID: '{{ $v | get "azure.keyVault.tenantId" ($v | get "azure.tenantId" "") }}'
-    AZURE_CLIENT_ID: '{{ $v | get "azure.keyVault.clientId" "" }}'
-    AZURE_CLIENT_SECRET: '{{ $v | get "azure.keyVault.clientSecret" "" }}'
-    GCLOUD_SERVICE_KEY='{{ $v | get "google.kmsAccount" "" | replace "\n" "" }}'
+
+    {{- if hasKey $kms "azurekeyvault" }}
```
Comment (Morriz) [inline values/drone/drone.gotmpl:70]:
```suggestion
```

**Review by Morriz — CHANGES_REQUESTED** [review]

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Thanks for enhancing settings for platform secrets.
Could you tell us, how did you test it?
So far a minor change that need to be applied in many places.

**Comment by Morriz** [comment]
It starts to annoy me to see "Thank you" all the time for every little thing that is part of the work we do, and we get salaries for. I would not formalize this but instead keep it special for when somebody goes _beyond_ the expected.

But maybe that is me being allergic to every form of communication that loses meaning or impact over time.

Just wanted to share that.

**Comment by j-zimnowoda** [comment]
> It starts to annoy me to see "Thank you" all the time for every little thing that is part of the work we do, and we get salaries for. I would not formalize this but instead keep it special for when somebody goes _beyond_ the expected.
> 
> But maybe that is me being allergic to every form of communication that loses meaning or impact over time.
> 
> Just wanted to share that.

I am afraid that it is your allergy and is not relevant to the content go the PR. No value added, no comment please.

File: values/drone/drone.gotmpl [hunk 4]
```diff
@@ -67,31 +67,25 @@ kubernetes:
...
-    {{- if hasKey $kms "azurekeyvault" }}
-    {{- with $kms.azurekeyvault }}
-    AZURE_TENANT_ID: '{{- .tenant_id -}}'
-    AZURE_CLIENT_ID: '{{- .client_id -}}'
-    AZURE_CLIENT_SECRET: '{{- .client_secret -}}'
+    {{- with ($kms | get "azurekeyvault" nil) }}
+    AZURE_TENANT_ID: '{{ .tenant_id }}'
+    AZURE_CLIENT_ID: '{{ .client_id }}'
+    AZURE_CLIENT_SECRET: '{{ .client_secret }}'
     AZURE_ENVIRONMENT: "AZUREPUBLICCLOUD"
```
Comment (mojtabaimani) [inline values/drone/drone.gotmpl:74]: I am not sure about this. I just copied it from the current code.

File: values/drone/drone.gotmpl [hunk 5]
```diff
@@ -67,31 +67,25 @@ kubernetes:
...
-    GOOGLE_PROJECT: '{{- .project -}}'
-    GOOGLE_REGION: '{{- .region -}}'
-    VAULT_GCPCKMS_SEAL_KEY_RING: '{{- .key_ring -}}'
-    GCLOUD_SERVICE_KEY: '{{- .kmsAccount -}}'
-    {{- end }}
+    {{- with ($kms | get "gcpckms" nil) }}
+    GOOGLE_PROJECT: '{{ .project }}'
+    GOOGLE_REGION: '{{ .region }}'
+    VAULT_GCPCKMS_SEAL_KEY_RING: '{{ .key_ring }}'
+    GCLOUD_SERVICE_KEY: '{{ .kmsAccount }}'
```
Comment (mojtabaimani) [inline values/drone/drone.gotmpl:88]: I am not sure about this.

**Review by svatwork — APPROVED** [review]

**Comment by j-zimnowoda** [comment]
please, merge with master

**Pull request merged**